### DISCUSSION TITLE
Does the Guild Event, Leave, work with bans and kicks?

### DISCUSSION THREAD

**Discussion by Pluppers** [discord-jda/JDA#2149] [Questions and Help] [answered]
Context: I am creating a file for each and every person on my server, and if someone leaves, I want to delete their file to save some space. I want to know if the guild leave event also counts kicks and bans so that I can work around it if it doesn't.

**Comment by MinnDevelopment** [comment, marked as answer]
The [GuildMemberRemoveEvent](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/events/guild/member/GuildMemberRemoveEvent.html) is fired for any kind of way a member is removed from a guild. This includes kick/ban/leave just like the docs say.

**Reply by Pluppers** [reply]
Thank you!
